src/server/tran_server.cpp

```diff
@@ -373,10 +372,25 @@ tran_server::connection_handler::connect ()
   // Do the preliminary jobs depending on the server type before opening the conneciton to the outisde.
   on_connecting ();
 
-  auto lockg_state = std::lock_guard<std::shared_mutex> { m_state_mtx };
-  assert (m_state == state::CONNECTING);
+  {
+    auto lockg_state = std::lock_guard<std::shared_mutex> { m_state_mtx };
+    assert (m_state == state::CONNECTING);
 
-  m_state = state::CONNECTED;
+    m_state = state::CONNECTED;
+  }
+  /*
+   *  Now, the main connection is reset whenever a new connection is established.
+   *  The main connection information is used in the following connection_handler's connection.
+   *  Sepcifically, the target PS to catch up with during CONNECTING is the main connection.
+   *
+   *  TODO: later, when the ATS recovery with multi-PS comes in, a leader PS elected will be the target PS,
+   *  and the main connection doesn't need to be set here, but in more appropriate location.
+   *  For example, it can be set after establishing all connection.
+   */
+  if (m_ts.reset_main_connection () != NO_ERROR)
```

Comment:
My opinion is that this call should be taken out of here.
And:
- in `tran_server::init_page_server_hosts` be called only once after all connections are established - as it was before 
- in `ps_connector::try_connect_to_all_ps` be called after each successful connection 

I see a clear semantic difference between
- connection_handler::connect to a peer server
- main connection reset, handled by whoever calls the connection_handler::connect (a higher hierarchical entity than the connection_handler)

Or, in other words, "do one thing, and do it well. :)
Feel free to ignore this suggestion. :)

Also:
BTW `ps_connector::try_connect_to_all_ps` does not interpret the code from `connection_handler::connect` - either change to
```
(void) conn->connect ();
```
or handle the error code.

Reply:
> My opinion is that this call should be taken out of here.

That's the same as what @hornetmj pointed out and I agree. However, the main connection should be set during the first `connect()` for the following `connect()`. See the comment:
```
  *  Now, the main connection is reset whenever a new connection is established.
    *  The main connection information is used in the following connection_handler's connection.
    *  Sepcifically, the target PS to catch up with during CONNECTING is the main connection.
```

I can probably get it done in another way. For example, we can send the catch-up request after all `connect()` are done, and do the state transition to CONNECTED. However, the whole connection sequence will be modified in the next epic (ATS recovery). It's the simplest fix before then.

Reply:
Fair enough.